drivers/modem/modem_cellular.c

```diff
@@ -88,11 +89,17 @@ struct modem_cellular_data {
 	uint8_t cmux_transmit_buf[2 * CONFIG_MODEM_CELLULAR_CMUX_MAX_FRAME_SIZE];
 	struct modem_cmux_dlci dlci1;
 	struct modem_cmux_dlci dlci2;
+	struct modem_cmux_dlci dlci3;
+	struct modem_cmux_dlci dlci4;
 	struct modem_pipe *dlci1_pipe;
 	struct modem_pipe *dlci2_pipe;
+	struct modem_pipe *dlci3_pipe;
+	struct modem_pipe *dlci4_pipe;
 	uint8_t dlci1_receive_buf[CONFIG_MODEM_CELLULAR_CMUX_MAX_FRAME_SIZE];
 	/* DLCI 2 is only used for chat scripts. */
 	uint8_t dlci2_receive_buf[CONFIG_MODEM_CELLULAR_CHAT_BUFFER_SIZES];
+	uint8_t dlci3_receive_buf[CONFIG_MODEM_CELLULAR_CHAT_BUFFER_SIZES];
+	uint8_t dlci4_receive_buf[CONFIG_MODEM_CELLULAR_CHAT_BUFFER_SIZES];
```

Comment:
We should likely do something about the buffer sizes... At least this can't be used like that when extending the number of DLCIs and their use case.

Reply:
A new config named MODEM_CELLULAR_USER_PIPE_BUFFER_SIZES has been added for the generic user pipes